meetings/2020/WASI-10-22.md

```diff
@@ -44,3 +44,161 @@ Installation is required, see the calendar invite.
     4. Other ideas?
 
 ## Meeting Notes
+
+### Attendees
+
+- Sam Clegg
+- Thomas Lively
+- Martin Duke
+- Syrus Akbary
+- Pat Hickey
+- Tanya Crenshaw
+- Alon Zakai
+- Andrew Brown
+- Mingqiu Sun
+- Alex Crichton
+- Dan Gohman
+- Steven Dabell
+- David Piepgrass
+- Johnnie Birch
+- Martin Duke
+- Yong
+- Sergey Rubanov
+- Arun Purushan
+
+### Find volunteers for note taking
+
+TL volunteers
+
+### Adoption of the agenda
+
+PH seconds
+
+### Notes
+
+#### Review of action items from prior meeting
+
+DG: Next time ??? is in the meeting we will check up on ???
+
+#### Vote to move wasi-nn to stage 2
+
+- Phases: https://github.com/WebAssembly/WASI/blob/master/docs/Process.md
+
+AB: We proposed wasi-nn a while back. Have a witx spec. I’ve been creating a POC
+to see if this works. When looking at docs, realized we are essentially in stage
+2. Moving to phase 2 would reflect reality. Asking for feedback on API.
+
+SC: English spec text requirement doesn’t apply?
+
+DG: Right, we only have witx.
+
+AB: Also generates md, which might count.
+
+SC: Implementation and test writing starts in stage 2. Probably don’t need a
+full vote. Does anyone have any comments or objections?
+
+PH: I followed it and I’m happy with how it looks.
+
+SC: Unanimous consent achieved to move to stage 2.
+
+AB: I will submit a PR updating which table it is in.
+
+#### Character encodings
+
+DG: Several threads about encodings going on. Want feedback about direction.
+
+<DG presenting> TODO: Add link to slides
+
+SC: Won’t the C program truncate on strcpy?
+
+DG: Yes, but then you’ll get a file not found error. <Back to slides>
+
+SC: Where does this ARF encoding come from?
+
+DG: I invented this after a lot of tries.
+
+SA: Is this for filenames only or also for contents?
+
+DG: File names, env vars, commandline args, but not file contents.
+
+SC: Anywhere a known string crosses the WASI boundary?
+
+DG: There might be other places. Where invalid strings should be roundtrippable without trapping. Some APIs trap, others wouldn’t. Some tools will want different things as well.
+
+TL: … userspace virtualization… ?
+
+PH: When IT describe interfaces, it would be great to use strings for these types, and IT guarantees unicode. ARF lets us use the IT string type.
+
+AC: In windows, filenames are lists of 16-bit values. How do we go from 16-bit to 8-bit strings? We need an extra layer.
+
+DG: We can extend ARF to handle Windows character space as well. ARF would be a little different on Windows and Linux.
+
+SC: missed this
```

Comment:
@sbc100 I missed what you said here. If you remember, it would be good to fill in.